e2e/backend_service_update_test.go

```diff
@@ -0,0 +1,237 @@
+	bsMutResource.Access(func(x *compute.BackendService) {
+		x.LoadBalancingScheme = lbScheme
+		x.Protocol = "TCP"
+		x.PortName = "http"
+		x.SessionAffinity = "NONE"
+		x.Port = 80
+		x.TimeoutSec = 30
+		x.ConnectionDraining = &compute.ConnectionDraining{}
+		x.HealthChecks = []string{hcID.SelfLink(meta.VersionGA)}
+	})
+	bsResource, err := bsMutResource.Freeze()
+	if err != nil {
+		return nil, err
+	}
+
+	bsBuilder := backendservice.NewBuilder(bsID)
+	bsBuilder.SetOwnership(rnode.OwnershipManaged)
+	bsBuilder.SetState(rnode.NodeExists)
+	bsBuilder.SetResource(bsResource)
+
+	graphBuilder.Add(bsBuilder)
+	return bsID, nil
+}
+
+func checkTCPRoute(t *testing.T, ctx context.Context, cloud cloud.Cloud, tcprID *cloud.ResourceID, rulesToBs [][]string) {
+	t.Helper()
+	t.Log("---- Check TCP Route ---- ")
+	tcpr, err := cloud.TcpRoutes().Get(ctx, tcprID.Key)
+	if err != nil {
+		t.Fatalf("cloud.TCPRoutes().Get(_, %s) = %v, want nil", tcprID.Key, err)
+	}
+	less := func(x, y string) bool { return x < y }
+	for i, r := range tcpr.Rules {
+		dests := r.Action.Destinations
+		srvcs := make([]string, len(dests))
+		for i, dst := range dests {
+			srvcs[i] = dst.ServiceName
+		}
+		if df := cmp.Diff(srvcs, rulesToBs[i], cmpopts.SortSlices(less)); df != "" {
+			t.Fatalf("Rule %d with action %+v  points to incorrect backend services, diff: %s , want nil", i, r.Action, df)
+		}
+	}
+}
+
+func checkBackendService(t *testing.T, ctx context.Context, cloud cloud.Cloud, bsID *cloud.ResourceID, bsPort int) {
+	t.Helper()
+	t.Log("---- Check BackendService ----")
+	gotBS, err := cloud.BackendServices().Get(ctx, bsID.Key)
+	if err != nil {
+		t.Fatalf("cloud.HealthChecks().Get(_, %s) = %v, want nil", bsID.Key, err)
+	}
+
+	if gotBS.Port != int64(bsPort) {
+		t.Errorf("BackendService port mismatch, got: %d want %d", gotBS.Port, bsPort)
+	}
+}
+func TestBackendServiceUpdate(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	graphBuilder := rgraph.NewBuilder()
+
+	meshURL, meshKey := ensureMesh(ctx, t)
+	t.Cleanup(func() {
+		err := theCloud.Meshes().Delete(ctx, meshKey)
+		t.Logf("theCloud.Meshes().Delete(ctx, %s): %v", meshKey, err)
+	})
+
+	hc1ID, err := buildHealthCheck(graphBuilder, "hc1-test", 15)
+	if err != nil {
+		t.Fatalf("buildHealthCheck(_, hc1-test, 15) = (_, %v), want (_, nil)", err)
+	}
+	hc2ID, err := buildHealthCheck(graphBuilder, "hc2-test", 15)
+	if err != nil {
+		t.Fatalf("buildHealthCheck(_, hc2-test, 15) = (_, %v), want (_, nil)", err)
+	}
+
+	bs1Name := "bs1-e2e"
+	bs2Name := "bs2-e2e"
+
+	bs1ID, err := buildBackendServiceWithLBScheme(graphBuilder, bs1Name, hc1ID, "INTERNAL_SELF_MANAGED")
+	if err != nil {
+		t.Fatalf("buildBackendServiceWithLBScheme(_, %s, _) = %v, want nil", resourceName(bs1Name), err)
+	}
+
+	bs2ID, err := buildBackendServiceWithLBScheme(graphBuilder, bs2Name, hc2ID, "INTERNAL_SELF_MANAGED")
+	if err != nil {
+		t.Fatalf("buildBackendServiceWithLBScheme(_, %s, _) = %v, want nil", resourceName(bs2Name), err)
```

Comment:
resourceName adds a prefix to name for cleaning up the resources after test (to prevent the leak). You should use this to set up the name  `bs1Name := resourceName("bs1-e2e")` and later use bs1Name everywhere.
If you use ` resourceName` here I recommend to use it for every resource in the test.

Reply:
I can see that is added to name in build... functions.

Reply:
Done.